test/common.py

```diff
@@ -635,9 +638,19 @@ def metafunc(self, standalone):
         self.emcc_args.append('-Wno-unused-command-line-argument')
         # if we are impure, disallow all wasm engines
         if impure:
-          self.wasm_engines = []
-        nodejs = self.require_node()
-        self.node_args += shared.node_bigint_flags(nodejs)
+          wasm_engines = []
+        else:
+          wasm_engines = [engine for engine in self.wasm_engines
+            if all(excluded not in os.path.basename(engine[0]) for excluded in exclude_engines)]
```

Comment:
I find these double nested comprehension quite hard to understand.  Can we expand into a loop or split in to two statements perhaps?

How about:

```
wasm_engines = []
if not impure:
  for engine in self.wasm_engines:
     basename = os.path.basename(engine[0])
     if not any(pattern in basename for pattern in excluded_engines):
       wasm_engines.append(engine)
```

Reply:
Agreed, this is much more readable!